### ISSUE TITLE
Proposal: migarte `upload-api` into w3protocol repo

### ISSUE THREAD

**Issue by Gozala** [web3-storage/w3infra#127]
I think it would be much better if we were to move https://github.com/web3-storage/w3infra/tree/main/upload-api/service into https://github.com/web3-storage/w3protocol/tree/main/packages as `upload-api` package which:

1. Had no dependency on any of the AWS pieces
2. At construction were injected context with all dependencies providing interface for persistence.

For what it's worth that is how original reference implementation used to be before it was handed of to the trigram team 

https://github.com/web3-storage/ucanto/blob/w3/w3/store/src/store/provider.js#L7-L78

It also made it possible to test use in-memory stores in unit tests, which was really nice

**Comment by vasco-santos** [comment]
@Gozala yes that is the plan! https://github.com/web3-storage/w3infra/issues/84

We are currently blocked on having sst/seedrun and pnpm, but we probably can create our own release setup manually to unblock it. Just not a priority for MVP

**Comment by alanshaw** [comment]
This is complete yes?
